helper-script

```diff
@@ -0,0 +1,270 @@
+    moddir="$scriptdir/truckersmp"
+elif [ -z "$moddir" ]; then
+    moddir="$XDG_CACHE_HOME/truckersmp-cli"
+fi
+if [ -z "$protondir" ]; then
+    protondir=$(find_proton)
+fi
+
+# check for errors
+{
+# what to do?
+if ! [ "$update" = true ] && ! [ "$startgame" = true ]; then
+    >&2 echo "Update (-u) and/or start the game (-s)?"
+fi
+
+# need account to update
+if [ "$update" = true ] && [ -z "$accountname" ]; then
+    >&2 echo "Need the steam account name to update. (-n name)"
+    exit 1
+fi
+
+# only one game at a time
+if [ "$ats" = true ] && [ "$ets2" = true ]; then
+    >&2 echo "It's only possible to use one game at a time."
+    exit 1
+elif ! [ "$ats" = true ] && ! [ "$ets2" = true ]; then
+    >&2 echo "Need at least one game. ATS (-a) or ETS2 (-e)"
+    exit 1
+elif [ "$ats" = true ]; then
+    steamid="270880"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/data"
+    fi
+elif [ "$ets2" = true ]; then
+    steamid="227300"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/data"
+    fi
+fi
+
+echo "Using $steamid"
+echo "GAMEDIR: $gamedir"
+echo "prefixdir: $prefixdir"
+echo "protondir: $protondir"
+
+# check $protondir
+if [ "$startgame" = true ] &&  ! [ -f "$protondir/proton" ]; then
+    >&2 echo "Proton not found."
+    >&2 echo    "Make sure to start at least one game with proton manually or" \
+                "set the correct path with -p"
+    exit 1
+fi
+
+# check for game if no update
+if  ! [ "$update" = true ] \
+    && ! [ -f "$gamedir/bin/win_x64/eurotrucks2.exe" ] \
+    && ! [ -f "$gamedir/bin/win_x64/amtrucks.exe" ]; then
+    >&2 echo "No update wanted and game not found."
+    exit 1
+fi
+}
+
+# download ATS/ETS2
+if [ "$update" = true ]; then
+    # make sure steam is closed before updating
+    # it's possible to update with the steam client open but the client looses
+    # all connectivity and asks for password and steam guard code after restart
+    while pidof steam; do
+	    pkill steam
+	    sleep 5
+    done
+
+    # get steamcmd if not already available
+    if ! [ -f "$moddir/steamcmd/steamcmd.sh" ]; then
+        mkdir -p "$moddir/steamcmd"
+        wget -qO- $steamcmdurl | tar xvz -C "$moddir/steamcmd"
+    fi
+
+    if ! [ -d "$gamedir" ]; then
+        mkdir -p "$gamedir"
+    fi
+
+    "$moddir/steamcmd/steamcmd.sh" \
+        +@sSteamCmdForcePlatformType windows \
+        +login "$accountname" \
+        +force_install_dir "$gamedir" \
+        +app_update "$steamid" validate \
+        +quit
+
+    # update the script itself
+    if git -C "$scriptdir" rev-parse --is-inside-work-tree > /dev/null 2>&1; then
```

Comment:
This one tries to break everything when you're not inside master. We could test for that
```suggestion
    if git -C "$scriptdir" rev-parse --is-inside-work-tree > /dev/null 2>&1 && [ "$(git -C "$scriptdir" branch --list | awk '/\*/{print $2})" = master ]; then
```

Reply:
Can't follow you here. I'm working inside a branch and can use this command without problems to determine if the directory is in a git environment.
And if it's version controlled with git we pull from origin master which is also possible from another branch. But it could be that there is no origin master, I should catch that.

Reply:
While I was trying your script, I had checked out the steam_play_proton branch, and in updating the repo it tried to merge the branch unto master (I believe that's what it was).
That's the kind of problem I was worried about: messing up someone custom branch.

Reply:
Maybe we should drop this and let the user do this on it's own?

Reply:
No had the right idea, self updating is a great feature, especially with users who might not be very tech-savvy.
I think disabling auto-update when master isn't checked out could work, maybe with a warning.